numba/parfors/parfor.py

```diff
@@ -1763,7 +1763,27 @@ def run(self, blocks):
                         else:
                             # Handle A[:] = x
                             shape = equiv_set.get_shape(instr)
-                            if shape is not None:
+                            # Don't converted broadcasted setitems into parfors.
+                            if isinstance(index_typ, types.BaseTuple):
+                                # The sliced dims are those in the index that
+                                # are made of slices.  Count the numbers of slices
+                                # in the index tuple.
+                                sliced_dims = len(list(filter(
+                                    lambda x: isinstance(x, types.misc.SliceType),
+                                    index_typ.types)))
+                            elif isinstance(index_typ, types.misc.SliceType):
+                                # For singular indices there can be a bare slice
+                                # and if so there is one dimension being set.
+                                sliced_dims = 1
+                            else:
+                                sliced_dims = 0
```

Comment:
Are these branches exercised by the tests?

Reply:
I checked and all 3 branches are covered by the test suite.

Reply:
Thanks for checking, I can see the same locally. The change in the logic to cover the `setitem` usecases seems valid on inspection.